libs/blocks/email-collection/utils.js

```diff
@@ -0,0 +1,271 @@
+import { createTag, getConfig, getFederatedUrl, localizeLink } from '../../utils/utils.js';
+
+const ECID_COOKIE = 'AMCV_9E1005A551ED61CA0A490D45@AdobeOrg';
+const API_ENDPOINTS = {
+  stage: 'https://www.stage.adobe.com/milo-email-collection-api',
+  prod: 'https://www.adobe.com/milo-email-collection-api',
+};
+const FORM_METADATA = {
+  'mps-sname': 'mpsSname',
+  'subscription-name': 'subscriptionName',
+};
+
+export function localizeFederatedUrl(url) {
+  return localizeLink(getFederatedUrl(url), '', true);
+}
+
+const FEDERAL_ROOT = '/federal/email-collection';
+const CONSENT_URL = localizeFederatedUrl(`${FEDERAL_ROOT}/consents/cs4.plain.html`);
+const PLACEHOLDER_URL = localizeFederatedUrl(`${FEDERAL_ROOT}/form-config.json?sheet=placeholders`);
+
+export const FORM_FIELDS = {
+  email: {
+    tag: 'input',
+    attributes: {
+      type: 'email',
+      required: true,
+      autocomplete: 'email',
+    },
+  },
+  'first-name': {
+    tag: 'input',
+    attributes: {
+      type: 'text',
+      readonly: '',
+    },
+  },
+  'last-name': {
+    tag: 'input',
+    attributes: {
+      type: 'text',
+      readonly: '',
+    },
+  },
+  country: {
+    tag: 'input',
+    url: localizeFederatedUrl(`${FEDERAL_ROOT}/form-config.json?sheet=countries`),
+    attributes: {
+      type: 'text',
+      readonly: '',
+    },
+  },
+  organization: {
+    tag: 'input',
+    attributes: {
+      type: 'text',
+      required: true,
+      autocomplete: 'organization',
+    },
+  },
+  occupation: {
+    tag: 'input',
+    attributes: {
+      type: 'text',
+      required: true,
+      autocomplete: 'organization-title',
+    },
+  },
+  state: {
+    tag: 'select',
+    url: getFederatedUrl(`${FEDERAL_ROOT}/form-config.json?sheet=states&limit=2000`),
+    attributes: { required: true },
+  },
+};
+
+export async function getIMS() {
+  if (window.adobeIMS) return window.adobeIMS;
```

Comment:
You could use `loadIms` from [utils](https://github.com/adobecom/milo/blob/stage/libs/utils/utils.js#L1331), this will ensure that you get access to `adobeIMS` object.